src/scripts/main.js

```diff
@@ -1 +1,80 @@
 'use strict';
+
+async function wait(ms) {
+  return new Promise((resolve) => setTimeout(resolve, ms));
+}
+
+const firstPromise = new Promise(async (resolve, reject) => {
+  let isTrue = false;
+
+  document.addEventListener('click', () => {
+    isTrue = true;
+    resolve('First promise was resolved');
+  });
+
+  await wait(3000);
+
+  if (!isTrue) {
+    reject(new Error('First promise was rejected'));
+  }
+});
+
+const secondPromise = new Promise((resolve, reject) => {
+  let isTrue = false;
+
+  document.addEventListener('click', () => {
+    resolve('Second promise was resolved');
+    isTrue = true;
+  });
+
+  if (isTrue !== true) {
+    document.addEventListener('contextmenu', () => {
+      resolve('Second promise was resolved');
+    });
+  }
+});
+
+const thirdPromise = new Promise((resolve, reject) => {
+  let firstClicked = false;
+  let twoClicked = false;
+
+  document.addEventListener('click', () => {
+    firstClicked = true;
+
+    if (twoClicked === true && firstClicked === true) {
+      resolve('Third promise was resolved');
+    }
+  });
+
+  document.addEventListener('contextmenu', (e) => {
+    twoClicked = true;
+    e.preventDefault();
+
+    if (twoClicked === true && firstClicked === true) {
+      resolve('Third promise was resolved');
+    }
+  });
+});
```

Comment:
Bug: The thirdPromise will be resolved if the user clicks and then right-clicks, but it will also be resolved if the user right-clicks and then clicks. If you want the promise to be resolved only when the user clicks and then right-clicks (in that order), you should check that `firstClicked` is `true` inside the second event listener, and check that `twoClicked` is `false` inside the first event listener. This way, the promise will be resolved only when the user clicks and then right-clicks.